subxt/src/error/dispatch_error.rs

```diff
@@ -142,6 +142,26 @@ impl PartialEq for ModuleError {
 
 impl Eq for ModuleError {}
 
+/// Custom `Debug` implementation, ignores the very large `metadata` field, using it instead (as
+/// intended) to resolve the actual pallet and error names. This is much more useful for debugging.
+impl Debug for ModuleError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let details = match self.details() {
+            Ok(details) => format!(
+                "{pallet_name}::{variant_name}",
+                pallet_name = details.pallet.name(),
+                variant_name = details.variant.name,
+            ),
+            Err(_) => format!(
+                "Unknown pallet error '{bytes:?}' (pallet and error details cannot be retrieved)",
+                bytes = self.bytes
+            ),
+        };
+
+        write!(f, "ModuleError(<{details}>)")
```

Comment:
nit: Would it be beneficial to have this syntax `ModuleError<..>` also for the `Display` impl?

Reply:
Fine by me, wdyt @jsdw?

Reply:
I think the display impl can be "prettier" (it doesn't need to fit in with the rust like syntax that everything else spits out for debug impls), but perhaps we should indeed standardise these; maybe Debug should print `ModuleError(<{details}>)` as it does here and Display should just print `{details}`, so an inner func to write details and then Debug just wraps to make fit better with rusty syntax?

Reply:
Done